### PR TITLE
add context listeners to consul client

### PR DESCRIPTION
**Pull request by ghost**
@rebuy-de/it-platform @rebuy-de/prp-consul-java-client

### CODE REVIEW

File: src/main/java/com/rebuy/consul/listeners/ContextClosedEventListener.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package com.rebuy.consul.listeners;
+
+import com.rebuy.consul.ConsulService;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.event.ContextClosedEvent;
+
+public class ContextClosedEventListener implements ApplicationListener<ContextClosedEvent>
```
Comment (svenwltr) [inline src/main/java/com/rebuy/consul/listeners/ContextClosedEventListener.java:7]: These class names are very generic. May this cause ambiguity on using this in other projects?

File: pom.xml [hunk 2]
```diff
@@ -90,6 +92,12 @@
             <version>${slf4j.version}</version>
         </dependency>
         <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot</artifactId>
+            <version>${spring-boot.version}</version>
```
Comment (bjoernhaeuser) [inline pom.xml:97]: do we really need spring-boot or would spring-context be enough?

File: src/main/java/com/rebuy/consul/listeners/ContextRefreshedEventListener.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package com.rebuy.consul.listeners;
+
+import com.rebuy.consul.ConsulService;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.event.ContextRefreshedEvent;
+
+public class ContextRefreshedEventListener implements ApplicationListener<ContextRefreshedEvent>
+{
+    final private ConsulService consulService;
```
Comment (bjoernhaeuser) [inline src/main/java/com/rebuy/consul/listeners/ContextRefreshedEventListener.java]: `private final`

File: src/main/java/com/rebuy/consul/listeners/EmbeddedServletContainerInitializedEventListener.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package com.rebuy.consul.listeners;
+
+import com.rebuy.consul.ConsulService;
+import org.springframework.boot.context.embedded.EmbeddedServletContainerInitializedEvent;
+import org.springframework.context.ApplicationListener;
+
+public class EmbeddedServletContainerInitializedEventListener implements ApplicationListener<EmbeddedServletContainerInitializedEvent>
```
Comment (bjoernhaeuser) [inline src/main/java/com/rebuy/consul/listeners/EmbeddedServletContainerInitializedEventListener.java:7]: why do we need this?

File: src/main/java/com/rebuy/consul/listeners/ContextClosedEventListener.java (hunk 1 above)
Comment (ghost) [inline src/main/java/com/rebuy/consul/listeners/ContextClosedEventListener.java:7]: no it shouldn't we dont do a lot of event handling

File: pom.xml (hunk 2 above)
Comment (ghost) [inline pom.xml:97]: we need it for the EmbeddedServletContainerInitializedEvent

File: src/main/java/com/rebuy/consul/listeners/EmbeddedServletContainerInitializedEventListener.java (hunk 4 above)
Comment (ghost) [inline src/main/java/com/rebuy/consul/listeners/EmbeddedServletContainerInitializedEventListener.java:7]: even though the docu is old:
http://docs.spring.io/autorepo/docs/spring-boot/1.2.0.M2/api/org/springframework/boot/context/embedded/EmbeddedServletContainerInitializedEvent.html
it is called after refresh ready so it should be better.

**Comment by bjoernhaeuser** [comment]
:+1:

**Comment by svenwltr** [comment]
:+1:

**Pull request merged**